### PR TITLE
setting read/write http timeouts

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this pr sets read and write http timeouts. As by default these timeouts are off, it means that clients can hold up connections indefinitely this in turn can potentially use up all available connections in a pool.

### CODE REVIEW

File: api/cmd/kubermatic-cluster-controller/main.go [hunk 1]
```diff
@@ -121,8 +121,10 @@ func main() {
 		m.Handle(*prometheusPath, promhttp.Handler())
 
 		s := http.Server{
-			Addr:    *prometheusAddr,
-			Handler: m,
+			Addr:         *prometheusAddr,
+			Handler:      m,
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
```
Comment (p0lyn0mial) [inline api/cmd/kubermatic-cluster-controller/main.go:127]: set arbitrarily.

**Comment by p0lyn0mial** [comment]
@mrIncompetent, @metalmatze could you help identify other potential places in our codebase where this could be relevant.

**Comment by mrIncompetent** [comment]
The only place i could think of is the Dashboard-v2 go-server.
I don't think that we have other http.Server's

**Comment by metalmatze** [comment]
I'm not sure if we really need to set this on the server. Nevertheless I'm fine with setting a default.

More importantly, when doing HTTP calls from Go as a client, you should **always** set a timeout. :blush:

**Review by metalmatze — APPROVED** [review]

**Pull request merged**